--- a/Makefile	2014-12-21 02:18:28.000000000 +0100
+++ b/Makefile	2015-07-16 20:41:32.203180832 +0200
@@ -7,7 +7,6 @@
 # Uncomment the following line if you want EEE support completely removed.
 #TG3_EXTRA_DEFS += TG3_NO_EEE
 
-KVER=
 ifeq ($(KVER),)
 	KVER=$(shell uname -r)
 endif
@@ -18,7 +17,8 @@
 endif
 
 ifeq ($(BCMCFGDIR),)
-	BCMCFGS := /lib/modules/$(KVER)/build \
+	BCMCFGS := $(KSRC) \
+                   /lib/modules/$(KVER)/build \
 	           /usr/src/kernels/$(KVER)-$(BCMPROC) \
 	           /usr/src/linux-$(shell echo $(KVER) | sed "s|\(.*\)-\([a-z]*\)|\1-obj/$(shell uname -i)/\2|") \
 	           /usr/src/linux
@@ -35,7 +35,8 @@
 endif
 
 ifeq ($(BCMSRCDIR),)
-	BCMSRCS := /lib/modules/$(KVER)/source \
+	BCMSRCS := $(KSRC) \
+                   /lib/modules/$(KVER)/source \
 	           /lib/modules/$(KVER)/build  \
 	           /usr/src/kernels/$(KVER)-$(BCMPROC) \
 	           /usr/src/linux-$(shell echo $(KVER) | sed "s|\(.*\)-[a-z]*|\1|") \
@@ -97,7 +98,9 @@
 
 BCM_DRV = tg3.o
 
-CC = gcc
+ifeq ($(CC),)
+	CC = gcc
+endif
 
 CFLAGS=-DMODULE -D__KERNEL__ -Wall -Wstrict-prototypes -I$(BCMCFGDIR)/include -fno-strict-aliasing
 
@@ -105,7 +108,9 @@
   CFLAGS+=-DOLD_NETIF
 endif
 
-ARCH:=$(shell uname -m)
+ifndef ARCH
+  ARCH:=$(shell uname -m)
+endif
 
 ifeq ($(ARCH),x86_64)
   CFLAGS+=-mno-red-zone -mcmodel=kernel -pipe -finline-limit=2000
