--- Makefile.orig	2014-10-25 11:49:45.000000000 +0200
+++ Makefile	2015-04-16 09:43:03.909638903 +0200
@@ -50,13 +50,13 @@
 	ln -f $< $@
 
 advancecomp-1.15/advfs:
-	( cd advancecomp-1.15 ; ./configure && $(MAKE) advfs )
+	( cd advancecomp-1.15 ; ARCH=i386 ./configure --host=i386-linux-gnu && $(MAKE) advfs )
 
 extract_compressed_fs: extract_compressed_fs.c
-	$(CC) -Wall -O2 $(CFLAGS) $(LDFLAGS) -o $@ $< -lz
+	$(CC) -m32 -Wall -O2 $(CFLAGS) $(LDFLAGS) -o $@ $< -lz
 
 cloop_suspend: cloop_suspend.o
-	$(CC) -Wall -O2 -s -o $@ $<
+	$(CC) -m32 -Wall -O2 -s -o $@ $<
 
 clean:
 	rm -rf create_compressed_fs extract_compressed_fs zoom *.o *.ko Module.symvers .cloop* .compressed_loop.* .tmp* modules.order cloop.mod.c
