--- Makefile.orig	2010-05-13 23:07:51.000000000 +0200
+++ Makefile	2014-10-13 16:22:26.194288447 +0200
@@ -123,7 +123,7 @@
 	install -D -m 644 $(MAN)/$(*F) $@
 
 $(BIN)/%: $(OBJS)
-	$(CC) -o $@ $^ $(LDFLAGS)
+	$(CC) -m32 -o $@ $^ $(LDFLAGS)
 
 $(MESSAGES): $(SRC_FILES) $(INC_FILES)
 	xgettext -k_ -o$@ $^
@@ -131,16 +131,16 @@
 $(OBJS): $(OBJ)/%.o: $(SRC)/%.c $(DEP)/%.d \
                       $(filter-out $(wildcard $(OBJ)), $(OBJ)) \
                       $(filter-out $(wildcard $(BIN)), $(BIN))
-	$(CC) $(CFLAGS) -o $@ $<
+	$(CC) -m32 $(CFLAGS) -o $@ $<
 
 $(MO_FILES): $(MO)/%.mo: $(PO)/%.po $(filter-out $(wildcard $(MO)), $(MO))
 	msgfmt -o $@ $<
 
 $(DEPS): $(DEP)/%.d: $(SRC)/%.c $(filter-out $(wildcard $(DEP)), $(DEP))
 ifeq ($(MAKECMDGOALS),quiet)
-	@$(CC) -MM $(CFLAGS) -MT $@ $< > $@
+	@$(CC) -m32 -MM $(CFLAGS) -MT $@ $< > $@
 else
-	$(CC) -MM $(CFLAGS) -MT $@ $< > $@
+	$(CC) -m32 -MM $(CFLAGS) -MT $@ $< > $@
 endif
 
 ifneq ($(MAKECMDGOALS),mrproper)
