diff -ur cloop-3.14.1.2.orig/cloop.c cloop-3.14.1.2/cloop.c
--- cloop-3.14.1.2.orig/cloop.c	2015-05-03 13:22:21.000000000 +0200
+++ cloop-3.14.1.2/cloop.c	2016-05-07 19:01:51.548280550 +0200
@@ -460,17 +460,6 @@
  /* In suspended mode, we have done all checks necessary - FF */
  if (clo->suspended)
    return error;
- if(isblkdev)
-  {
-   struct request_queue *q = bdev_get_queue(inode->i_bdev);
-   blk_queue_max_hw_sectors(clo->clo_queue, queue_max_hw_sectors(q)); /* Renamed in 2.6.34 */
-   blk_queue_max_segments(clo->clo_queue, queue_max_segments(q)); /* Renamed in 2.6.34 */
-   /* blk_queue_max_hw_segments(clo->clo_queue, queue_max_hw_segments(q)); */ /* Removed in 2.6.34 */
-   blk_queue_max_segment_size(clo->clo_queue, queue_max_segment_size(q));
-   blk_queue_segment_boundary(clo->clo_queue, queue_segment_boundary(q));
-   blk_queue_merge_bvec(clo->clo_queue, q->merge_bvec_fn);
-   clo->underlying_blksize = block_size(inode->i_bdev);
-  }
  else
    clo->underlying_blksize = PAGE_SIZE;
  DEBUGP("Underlying blocksize is %u\n", clo->underlying_blksize);
