diff -up cloop-3.14.1.2/cloop.c.orig cloop-3.14.1.2/cloop.c
--- cloop-3.14.1.2/cloop.c.orig	2016-12-25 14:59:52.000000000 +0100
+++ cloop-3.14.1.2/cloop.c	2016-12-25 20:48:31.960426883 +0100
@@ -410,11 +410,13 @@ static void cloop_do_request(struct requ
    if (unlikely(req == NULL || (req->cmd_type != REQ_TYPE_FS)))
     goto error_continue;
    rw = rq_data_dir(req);
+/*
    if (unlikely(rw != READ && rw != READA))
     {
      DEBUGP("cloop_do_request: bad command\n");
      goto error_continue;
     }
+*/
    clo = req->rq_disk->private_data;
    if (unlikely(!clo->backing_file && !clo->suspended))
     {
