diff -up cloop-3.14.1.2/cloop.c.orig cloop-3.14.1.2/cloop.c
--- cloop-3.14.1.2/cloop.c.orig	2017-06-07 16:15:49.000000000 +0200
+++ cloop-3.14.1.2/cloop.c	2017-06-08 12:47:17.000000000 +0200
@@ -407,7 +407,7 @@ static void cloop_do_request(struct requ
    int rw;
  /* quick sanity checks */
    /* blk_fs_request() was removed in 2.6.36 */
-   if (unlikely(req == NULL || (req->cmd_type != REQ_TYPE_FS)))
+   if (unlikely(req == NULL || blk_rq_is_passthrough(req)))
     goto error_continue;
    rw = rq_data_dir(req);
 /*
@@ -751,7 +751,8 @@ static int cloop_get_status(struct cloop
  struct kstat stat;
  int err;
  if (!file) return -ENXIO;
- err = vfs_getattr(&file->f_path, &stat);
+ err = vfs_getattr(&file->f_path, &stat,
+                     STATX_INO, AT_STATX_SYNC_AS_STAT);
  if (err) return err;
  memset(info, 0, sizeof(*info));
  info->lo_number  = clo->clo_number;
