diff -up cloop-3.14.1.2/cloop.c.orig cloop-3.14.1.2/cloop.c
--- cloop-3.14.1.2/cloop.c.orig       2017-09-19 21:48:20.000000000 +0200
+++ cloop-3.14.1.2/cloop.c    2017-09-20 15:38:22.000000000 +0200
@@ -428,7 +428,7 @@ static void cloop_do_request(struct requ
    continue; /* next request */
   error_continue:
    DEBUGP(KERN_ERR "cloop_do_request: Discarding request %p.\n", req);
-   req->errors++;
+//   req->errors++; removed from linux/blkdev.h:request
    __blk_end_request_all(req, -EIO);
   }
 }

