diff -up ./qt-everywhere-opensource-src-4.8.6/src/corelib/arch/qatomic_i386.h.orig ./qt-everywhere-opensource-src-4.8.6/src/corelib/arch/qatomic_i386.h
--- ./qt-everywhere-opensource-src-4.8.6/src/corelib/arch/qatomic_i386.h.orig	2014-10-12 19:53:35.920469243 +0200
+++ ./qt-everywhere-opensource-src-4.8.6/src/corelib/arch/qatomic_i386.h	2014-10-12 19:57:04.984476107 +0200
@@ -169,7 +169,7 @@ Q_INLINE_TEMPLATE bool QBasicAtomicPoint
 {
     unsigned char ret;
     asm volatile("lock\n"
-                 "cmpxchgl %3,%2\n"
+                 "cmpxchgl %k3,%2\n"
                  "sete %1\n"
                  : "=a" (newValue), "=qm" (ret), "+m" (_q_value)
                  : "r" (newValue), "0" (expectedValue)
@@ -180,7 +180,7 @@ Q_INLINE_TEMPLATE bool QBasicAtomicPoint
 template <typename T>
 Q_INLINE_TEMPLATE T *QBasicAtomicPointer<T>::fetchAndStoreOrdered(T *newValue)
 {
-    asm volatile("xchgl %0,%1"
+    asm volatile("xchgl %k0,%1"
                  : "=r" (newValue), "+m" (_q_value)
                  : "0" (newValue)
                  : "memory");
