--- chntpw/Makefile.orig	2011-05-11 21:33:56.000000000 +0200
+++ chntpw/Makefile	2014-05-28 15:55:00.105959964 +0200
@@ -13,12 +13,12 @@
 CC=gcc
 
 # Force 32 bit
-CFLAGS= -DUSEOPENSSL -g -I. -I$(OSSLINC) -Wall -m32 
-OSSLLIB=$(OSSLPATH)/lib
+#CFLAGS= -DUSEOPENSSL -g -I. -I$(OSSLINC) -Wall -m32 
+#OSSLLIB=$(OSSLPATH)/lib
 
 # 64 bit if default for compiler setup
-#CFLAGS= -DUSEOPENSSL -g -I. -I$(OSSLINC) -Wall
-#OSSLLIB=$(OSSLPATH)/lib64
+CFLAGS= -DUSEOPENSSL -g -I. -I$(OSSLINC) -Wall
+OSSLLIB=$(OSSLPATH)/lib64
 
 
 # This is to link with whatever we have, SSL crypto lib we put in static
