Index: ssmtp-2.62/ssmtp.c
===================================================================
--- ssmtp-2.62.orig/ssmtp.c	2008-05-17 16:11:49.000000000 +0200
+++ ssmtp-2.62/ssmtp.c	2008-05-17 16:13:07.000000000 +0200
@@ -879,7 +879,7 @@
 		p=firsttok(&begin, "= \t\n");
 		if(p){
 			rightside=begin;
-			q = firsttok(&begin, "= \t\n:");
+			q = firsttok(&begin, "= \t\n");
 		}
 		if(p && q) {
 			if(strcasecmp(p, "Root") == 0) {
@@ -892,15 +892,15 @@
 				}
 			}
 			else if(strcasecmp(p, "MailHub") == 0) {
+				if((r = strchr(q, ':')) != NULL) {
+					*r++ = '\0';
+					port = atoi(r);
+				}
+			
 				if((mailhost = strdup(q)) == (char *)NULL) {
 					die("parse_config() -- strdup() failed");
 				}
 
-				if((r = firsttok(&begin, "= \t\n:")) != NULL) {
-					port = atoi(r);
-					free(r);
-				}
-
 				if(log_level > 0) {
 					log_event(LOG_INFO, "Set MailHub=\"%s\"\n", mailhost);
 					log_event(LOG_INFO, "Set RemotePort=\"%d\"\n", port);
