diff --git a/networking/udhcp/dhcpc.c b/networking/udhcp/dhcpc.c
index 9d3d1a31c..811a1a1ee 100644
--- a/networking/udhcp/dhcpc.c
+++ b/networking/udhcp/dhcpc.c
@@ -156,10 +156,6 @@ static const char *valid_domain_label(const char *label)
 	for (;;) {
 		ch = *label;
 		if ((ch|0x20) < 'a' || (ch|0x20) > 'z') {
-			if (pos == 0) {
-				/* label must begin with letter */
-				return NULL;
-			}
 			if (ch < '0' || ch > '9') {
 				if (ch == '\0' || ch == '.')
 					return label;
