--- rsync-3.0.6/xattrs.c	2009-04-26 16:51:50.000000000 +0200
+++ rsync-3.0.6-xattrs-system-patched/xattrs.c	2009-07-24 05:01:21.000000000 +0200
@@ -226,12 +226,14 @@
 		list_len -= name_len;
 
 #ifdef HAVE_LINUX_XATTRS
+#if 0
 		/* We always ignore the system namespace, and non-root
 		 * ignores everything but the user namespace. */
 		if (user_only ? !HAS_PREFIX(name, USER_PREFIX)
 			      : HAS_PREFIX(name, SYSTEM_PREFIX))
 			continue;
 #endif
+#endif
 
 		/* No rsync.%FOO attributes are copied w/o 2 -X options. */
 		if (name_len > RPRE_LEN && name[RPRE_LEN] == '%'
@@ -308,12 +310,14 @@
 		list_len -= name_len;
 
 #ifdef HAVE_LINUX_XATTRS
+#if 0
 		/* We always ignore the system namespace, and non-root
 		 * ignores everything but the user namespace. */
 		if (user_only ? !HAS_PREFIX(name, USER_PREFIX)
 			      : HAS_PREFIX(name, SYSTEM_PREFIX))
 			continue;
 #endif
+#endif
 
 		datum_len = 0;
 		if (!(ptr = get_xattr_data(source, name, &datum_len, 0)))
@@ -840,12 +844,14 @@
 		list_len -= name_len;
 
 #ifdef HAVE_LINUX_XATTRS
+#if 0
 		/* We always ignore the system namespace, and non-root
 		 * ignores everything but the user namespace. */
 		if (user_only ? !HAS_PREFIX(name, USER_PREFIX)
 			      : HAS_PREFIX(name, SYSTEM_PREFIX))
 			continue;
 #endif
+#endif
 		if (am_root < 0 && name_len > RPRE_LEN
 		 && name[RPRE_LEN] == '%' && strcmp(name, XSTAT_ATTR) == 0)
 			continue;
