--- a/cloop.c	2013-11-25 12:14:53.496688209 +0000
+++ b/cloop.c	2013-11-25 12:13:28.160691443 +0000
@@ -20,7 +20,7 @@
  */
 
 #define CLOOP_NAME "cloop"
-#define CLOOP_VERSION "2.639"
+#define CLOOP_VERSION "3.12"
 #define CLOOP_MAX 8
 
 #ifndef KBUILD_MODNAME
@@ -743,7 +743,7 @@
  struct kstat stat;
  int err;
  if (!file) return -ENXIO;
- err = vfs_getattr(file->f_path.mnt, file->f_path.dentry, &stat);
+ err = vfs_getattr(&file->f_path, &stat);
  if (err) return err;
  memset(info, 0, sizeof(*info));
  info->lo_number  = clo->clo_number;
